### ISSUE TITLE
Support copy-on-write mode for Iceberg write

### ISSUE THREAD

**Issue by jackye1995** [trinodb/trino#17272]
Today Iceberg writes only support merge-on-read mode. Copy-on-write mode is a frequent ask for users that want better file layout without the need to run compactions frequently.

Technically this could be achieved pretty easily. The CoW implementation is already available for Delta: 

https://github.com/trinodb/trino/blob/master/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

**Comment by findepi** [comment]
Yes, I think we should support copy-on-write mode for UPDATE and DELETE for case when we update/delete most of the file. Creating big delta files isn't helpful for anyone.
If we support this strategy, we should automatically choose between copy-on-write and merge-on-read within each `IcebergMergeSink` independently.

**Comment by C-h-e-r-r-y** [comment]
@rdblue 

For now these settings can not be set during table creation :(

**Comment by MotiPoyastro** [comment]
This feature is highly requested in my org,
Trino 443 + Iceberg on S3 parquets.
frequent merge into queries becomes a problem on a large scale.
currently dealing with post optimize/compaction for many small files created on a daily basis.
but this is not a desired flow for us.

**Comment by xkrogen** [comment]
One concern called out by @dain ([on Slack](https://trinodb.slack.com/archives/CJ6UC075E/p1729114539093519?thread_ts=1727902704.898979&cid=CJ6UC075E)) that I will persist here:

> CoW has it’s own problems because it breaks the change feed, so you cannot do downstream delta-style transforms (e.g., incremental materialized views)

I don't think this should be a blocker to adding support -- but something to keep in mind e.g. when considering doing a MoR vs CoW decision _automatically_ vs based on explicit user input, since there is some difference in functionality supported.

**Comment by hashhar** [comment]
With https://github.com/trinodb/trino/pull/24031 it should now be possible to specify the default operation mode via the table property as @rdblue mentioned above.

Is it still desirable to provide an "integrated" solution for this in Trino? for most use-cases I think the current default of MoR on Trino is good.

**Comment by autumnust** [comment]
@jackye1995  At this moment on the Iceberg side, we cannot set table properties to control the trino behavior on how it encodes delete files. Meanwhile https://iceberg.apache.org/docs/1.5.2/configuration/#write-properties didn't specify things like `write.delete.mode` is engine specific which is a bit confusing / inaccurate. 

Before this support lands on Trino, shall we fix the documentation on the Iceberg side as well?

**Comment by kaveti** [comment]
@MotiPoyastro @rdblue @jackye1995 @xkrogen @hashhar @autumnust can you please help on reviewing the PR : https://github.com/trinodb/trino/pull/28958  (support for COW in trino)

**Comment by kaveti** [comment]
gentle reminder @MotiPoyastro @rdblue @jackye1995 @xkrogen @hashhar @autumnust  on reviewing the PR: https://github.com/trinodb/trino/pull/28958

**Comment by kaveti** [comment]
gentle reminder @MotiPoyastro @rdblue @jackye1995 @xkrogen @hashhar @autumnust on reviewing the PR: https://github.com/trinodb/trino/pull/28958
